app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -102,6 +103,11 @@ def sap_profiles
     @sap_profiles ||= sap_api.pcloud_sap_getall(cloud_instance_id).profiles || []
   end
 
+  def server_groups
+    byebug
+    placement_groups_api.pcloud_placementgroups_getall(cloud_instance_id) || []
+  end
+
```

Comment:
I think the best way to link up vms to your server group is to build a hash of vm.id => server_group like this:
```ruby
def server_group_by_vm_id
  @server_group_by_vm_id ||= server_groups.each_with_object({}) { |sg, result| sg.members.each { |vm_id| result[vm_id] = sg } }
end
```

Then when you're parsing vms you can do:
```ruby
server_group_id = collector.server_group_by_vm_id[vm.id]&.id
server_group    = persister.server_groups.lazy_find(server_group_id) if server_group_id

persister.vms.build(
  ...
  :server_group => server_group
  ...
)
```